### PR TITLE
Added EL3104, EL3202, EL3318 Driver

### PR DESCRIPTION
**Pull request by davidinkyu-kim**
Added EL3104 driver
Tested online with test program

### CODE REVIEW

**Comment by davidinkyu-kim** [comment]
Issues resolving #5 #7 #8

**Comment by alex-brinkman** [comment]
- [ ] Update README with first version supported for these drivers

File: src/jsd_el3104.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  jsd_el3104_txpdo_t* txpdo =
+      (jsd_el3104_txpdo_t*)self->ecx_context.slavelist[slave_id].inputs;
+
+  int ch;
+  for (ch = 0; ch < JSD_EL3104_NUM_CHANNELS; ch++) {
+    state->adc_value[ch] = txpdo->channel[ch].value;
+
+    state->voltage[ch] =
+        (((double)state->adc_value[ch] + 32768) * 20 / 65536) - 10;
```
Comment (alex-brinkman) [inline src/jsd_el3104.c:36]: Let's use #defines for the conversion calculation constants. Can you confirm this expression simplifies down to just volts = adc / 3276.8 ?

Comment (alex-brinkman) [reply src/jsd_el3104.c]: resolved below

File: src/jsd_el3104_types.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * you will recieve stale data. As a check, you can use the TxPDO toggle field
+ * to see when the device has new data.
+ *
+ */
+typedef struct {
+  bool     limit1_enable[JSD_EL3104_NUM_CHANNELS];   ///< enables limit1
+  double   limit1_voltage[JSD_EL3104_NUM_CHANNELS];  ///< limit 1 in Volts
+  bool     limit2_enable[JSD_EL3104_NUM_CHANNELS];   ///< enables limit2
+  double   limit2_voltage[JSD_EL3104_NUM_CHANNELS];  ///< limit 2 in Volts
+} jsd_el3104_config_t;
```
Comment (alex-brinkman) [inline src/jsd_el3104_types.h:26]: These configuration parameters are not used in the setup PO2SO hook meaning they are dummy parameters. Either:
1. Remove them and remove `jsd_el3104_config_t` from the API altogether (EL2124 works this way)
2. Add full support for this feature

I recommend going with Option 1. If you pursue this option, please remove the `limit1` and `limit2` fields from the state struct.

File: src/jsd_el3104_types.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * @brief Read struct for el3104 module
+ */
+typedef struct {
+  double  voltage[JSD_EL3104_NUM_CHANNELS];    ///< Analog input data, converted
+  int16_t adc_value[JSD_EL3104_NUM_CHANNELS];  ///< Analog input data, raw
+  uint8_t limit1[JSD_EL3104_NUM_CHANNELS];     ///< 0-off, 1-exceeded, 2-under
+  uint8_t limit2[JSD_EL3104_NUM_CHANNELS];     ///< 0-off, 1-exceeded, 2-under
```
Comment (alex-brinkman) [inline src/jsd_el3104_types.h:35]: here

**Comment by alex-brinkman** [comment]
With the removal (or full implementation) of the Limits configuration and state variables, I approve the rest of the EL3104 module implementation.

**Comment by alex-brinkman** [comment]
An EL3202 module was tested with the 3-wire config with a PT100 and PT1000 and met expectations.
An EL3202-0100 module was tested with a 4-wire PT100 and met expectations. 
Based on what I'm seeing, I approve the EL3202 module implementation. 

David, did you perform any testing on the direct ohm measurements for this driver?

**Comment by alex-brinkman** [comment]
I need to acquire thermocouple hardware to test the EL3318 module, I'll try to follow up next week. The implementation for the EL3318 module looks good to me.

**Comment by alex-brinkman** [comment]
I tested a K-type and T-type thermocouple on the EL3318 driver and it worked as expected. 

Once the config variable issue on the EL3104 is resolved, we are ready to merge.

**Comment by davidinkyu-kim** [comment]
> An EL3202 module was tested with the 3-wire config with a PT100 and PT1000 and met expectations.
> An EL3202-0100 module was tested with a 4-wire PT100 and met expectations.
> Based on what I'm seeing, I approve the EL3202 module implementation.
> 
> David, did you perform any testing on the direct ohm measurements for this driver?

No, I only tested with PT100, it's hard to secure all possible types of PRTs.

**Comment by davidinkyu-kim** [comment]
I removed EL3104 "Limit" config and state. Feel free to go ahead and merge.

**Comment by alex-brinkman** [comment]
The EL3104 changes look great to me. The README matches the semver release I wanted for these changes too. Great work!

**Pull request merged**